pupil_src/shared_modules/video_export/plugin_base/isolated_frame_exporter.py

```diff
@@ -0,0 +1,125 @@
+"""
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2018 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+
+import os
+from abc import ABCMeta
+from glob import glob
+
+import player_methods as pm
+from av_writer import AV_Writer
+from task_manager import ManagedTask
+from video_capture import File_Source, EndofVideoError
+from video_export.plugin_base.video_exporter import VideoExporter
+
+
+class IsolatedFrameExporter(VideoExporter, metaclass=ABCMeta):
+    """
+    A VideoExporter that exports a part or all of some video file and applies
+    a function process_frame to every frame.
+    """
+
+    def add_export_job(
+        self,
+        export_range,
+        export_dir,
+        input_name,
+        output_name,
+        process_frame,
+        export_timestamps,
+    ):
+        os.makedirs(export_dir, exist_ok=True)
+        self.logger.info("Exporting to {}".format(export_dir))
+
+        input_video_file = _find_video_file(self.g_pool.rec_dir, input_name)
+        output_video_file = os.path.join(export_dir, output_name + ".mp4")
+        task_args = (
+            input_video_file,
+            output_video_file,
+            export_range,
+            self.g_pool.timestamps,
+            process_frame,
+            export_timestamps,
+        )
+        task = ManagedTask(
+            _convert_video_file,
+            args=task_args,
+            heading="Export {} Video".format(input_name),
+            min_progress=0.0,
+            max_progress=100.0,
+        )
+        self.add_task(task)
+
+
+def _find_video_file(directory, name):
+    try:
+        return [
+            f
+            for f in glob(os.path.join(directory, name + ".*"))
+            if os.path.splitext(f)[-1] in (".mp4", ".mkv", ".avi", ".mjpeg")
+        ][0]
+    except IndexError:
+        raise FileNotFoundError("No Video " + name + " found")
+
+
+def _convert_video_file(
+    input_file,
+    output_file,
+    export_range,
+    world_timestamps,
+    process_frame,
+    export_timestamps,
+):
+    yield "Export video", 0.0
+    inpt = File_Source(EmptyGPool(), input_file)
```

Comment:
You are more concerned about the "misspelling" or the missing "source"? I would go for `input_` as PEP8 suggests to add a single underscore to avoid clashes with buildin names